packages/tracing/src/browser/metrics.ts

```diff
@@ -179,6 +214,63 @@ export class MetricsInstrumentation {
       // Do nothing if the browser doesn't support this API.
     }
   }
+
+  /** Starts tracking the First Input Delay on the current page. */
+  private _trackFID(): void {
+    // Based on reference implementation from https://web.dev/fid/#measure-fid-in-javascript.
+    // Use a try/catch instead of feature detecting `first-input`
+    // support, since some browsers throw when using the new `type` option.
+    // https://bugs.webkit.org/show_bug.cgi?id=209216
+    try {
+      // Keep track of whether (and when) the page was first hidden, see:
+      // https://github.com/w3c/page-visibility/issues/29
+      // NOTE: ideally this check would be performed in the document <head>
+      // to avoid cases where the visibility state changes before this code runs.
+      let firstHiddenTime = document.visibilityState === 'hidden' ? 0 : Infinity;
+      document.addEventListener(
+        'visibilitychange',
+        event => {
+          firstHiddenTime = Math.min(firstHiddenTime, event.timeStamp);
+        },
+        { once: true },
+      );
+
+      const updateFID = (entry: PerformanceEventTiming, po: PerformanceObserver): void => {
+        // Only report FID if the page wasn't hidden prior to
+        // the entry being dispatched. This typically happens when a
+        // page is loaded in a background tab.
+        if (entry.startTime < firstHiddenTime) {
+          const fidValue = entry.processingStart - entry.startTime;
+
+          logger.log('[Measurements] Adding FID (First Input Delay)');
+
+          // Report the FID value to an analytics endpoint.
+          this._measurements['fid'] = { value: fidValue };
+
+          const timeOrigin = msToSec(performance.timeOrigin);
+          const startTime = msToSec(entry.startTime as number);
+          this._measurements['mark.fid'] = { value: timeOrigin + startTime };
```

Comment:
Reordering lines to match style of other occurrences (log, measurement, mark):

```suggestion
          const fidValue = entry.processingStart - entry.startTime;
          const timeOrigin = msToSec(performance.timeOrigin);
          const startTime = msToSec(entry.startTime as number);
          logger.log('[Measurements] Adding FID (First Input Delay)');
          this._measurements['fid'] = { value: fidValue };
          this._measurements['mark.fid'] = { value: timeOrigin + startTime };
```